pkg/ccl/backupccl/show.go

```diff
@@ -448,12 +449,81 @@ you must pass the 'encryption_info_dir' parameter that points to the directory o
 						len(dest)))
 			}
 		}
+
+		if _, ok := opts[backupOptCheckFiles]; ok {
+			if err := checkBackupFiles(ctx, info, p.ExecCfg().DistSQLSrv.ExternalStorageFromURI,
+				p.User()); err != nil {
+				return err
+			}
+		}
 		return infoReader.showBackup(ctx, &mem, mkStore, info, p.User(), resultsCh)
 	}
 
 	return fn, infoReader.header(), nil, false, nil
 }
 
+// checkBackupFiles validates that each SST is in its expected storage location
+func checkBackupFiles(
+	ctx context.Context,
+	info backupInfo,
+	storeFactory cloud.ExternalStorageFromURIFactory,
+	user security.SQLUsername,
+) error {
+
+	checkLayer := func(layer int) error {
+		// TODO (msbutler): Right now, checkLayer opens stores for each backup layer. In 22.2,
+		// once a backup chain cannot have mixed localities, only create stores for full backup
+		// and first incremental backup.
+		defaultStore, err := storeFactory(ctx, info.defaultURIs[layer], user)
+		if err != nil {
+			return err
+		}
+		localityStores := make(map[string]cloud.ExternalStorage)
+
+		defer func() {
+			if err := defaultStore.Close(); err != nil {
+				log.Warningf(ctx, "close export storage failed %v", err)
+			}
+			for _, store := range localityStores {
+				if err := store.Close(); err != nil {
+					log.Warningf(ctx, "close export storage failed %v", err)
+				}
+			}
+		}()
+
+		for locality, uri := range info.localityInfo[layer].URIsByOriginalLocalityKV {
+			store, err := storeFactory(ctx, uri, user)
+			if err != nil {
+				return err
+			}
+			localityStores[locality] = store
+		}
+
+		for _, f := range info.manifests[layer].Files {
+			store := defaultStore
+			uri := info.defaultURIs[layer]
+			if _, ok := localityStores[f.LocalityKV]; ok {
+				store = localityStores[f.LocalityKV]
+				uri = info.localityInfo[layer].URIsByOriginalLocalityKV[f.LocalityKV]
+			}
+			fd, err := store.ReadFile(ctx, f.Path)
```

Comment:
if you don't actually read a byte from `fd` you might never nit the network depending on the implementation (i.e. it could be lazy and just wait for Read() to do its first read).

you could just read 1 byte or something, but I dunno, I'd probably use `Size()` instead of `ReadFile()` here, avoid the the need to Close() and then you could also, if check_files (or stat_files?) is specified, you can change the result columns to include an "file_size" column with the result?

Reply:
that would be backward incompatible for SHOW BACKUP users. Is that ok?

Reply:
alternatively, I could rename 'compressed_size' to 'compressed_bytes'

Reply:
if you only do it when check_files is passed, it isn't breaking; but I'm also not worried about breaking `show backup files` _header name_ compatibility since we never doc'ed it, right?

Reply:
Currently, this diff also affects default SHOW BACKUP (see below). I feel like its a bit clunky to rename size_bytes only when check_files is passed. 

```
demo@127.0.0.1:26257/movr> SELECT database_name, object_name, size_bytes, rows, compressed_size FROM [SHOW BACKUP FROM LATEST IN 'userfile:///foo' WITH check_files];
  database_name |        object_name         | size_bytes | rows | compressed_size
----------------+----------------------------+------------+------+------------------
  NULL          | system                     |       NULL | NULL |            NULL
  system        | users                      |        300 |    3 |           10104
  system        | zones                      |         37 |    1 |             957
  system        | settings                   |        336 |    5 |           10104
  system        | ui                         |          0 |    0 |               0
  system        | jobs                       |       6086 |   15 |           10104
  system        | locations                  |        271 |    5 |           10104
  system        | role_members               |        204 |    2 |           10104
  system        | comments                   |          0 |    0 |               0
  system        | role_options               |          0 |    0 |               0
  system        | scheduled_jobs             |        255 |    1 |           10104
  system        | database_role_settings     |          0 |    0 |               0
  NULL          | defaultdb                  |       NULL | NULL |            NULL
  defaultdb     | public                     |       NULL | NULL |            NULL
  NULL          | postgres                   |       NULL | NULL |            NULL
  postgres      | public                     |       NULL | NULL |            NULL
  NULL          | movr                       |       NULL | NULL |            NULL
  movr          | public                     |       NULL | NULL |            NULL
  movr          | users                      |       5011 |   50 |           10104
  movr          | vehicles                   |       3242 |   15 |           10104
  movr          | rides                      |     159387 |  500 |           58254
  movr          | vehicle_location_histories |      75918 | 1000 |          144745
  movr          | promo_codes                |     222973 | 1000 |          144745
  movr          | user_promo_codes           |        416 |    5 |           10104
(24 rows)

```

Reply:
it's just weird to say compressed size since sstables aren't always compressed -- it's a per-block decision based on savings. I it really is the _file_ size vs the logical data size.

Reply:
we could just have size_bytes + file_size

Reply:
aha. good to know. I'll go with `file_bytes` for a bit more symmetry and will have docs explain the difference between the two.